src/core/shape_inference/include/one_hot_shape_inference.hpp

```diff
@@ -43,14 +43,13 @@ void shape_infer(const OneHot* op,
                           "off_value input must be scalar.");
 
     auto& result_shape = output_shapes[0];
-    std::vector<int64_t> depth_vals;
-    bool depth_is_set = get_data_as_int64<T>(1, op, depth_vals, constant_data);
     if (indices_shape.rank().is_static()) {
-        // decide result rank
         result_shape = indices_shape;
         const auto indices_rank = indices_shape.rank().get_length();
         const auto axis = ov::normalize_axis(op, op->get_axis(), indices_rank + 1, -indices_rank - 1, indices_rank);
 
+        std::vector<int64_t> depth_vals;
+        bool depth_is_set = get_data_as_int64<T>(1, op, depth_vals, constant_data);
         if (depth_is_set) {
             int64_t depth_val = depth_vals[0];
```

Comment:
`depth_is_set` can be true but the `depth_vals` can be still empty.  Add check that `depth_vals` is not empty to safely access first element.

Reply:
Decided, to remove `get_data_as_int64` and use `get_data_as_shape` as a common approach.